src/shared/copy.h

```diff
@@ -30,6 +30,13 @@ typedef enum CopyFlags {
         COPY_GRACEFUL_WARN = 1 << 15, /* Skip copying file types that aren't supported by the target filesystem */
 } CopyFlags;
 
+typedef enum DenyType {
+        DENY_INODE,
```

Comment:
maybe add a third enum type here, DENY_DONT (and map it explicitly to zero). We wouldn't ever add an entry with that, but it has the nice effect that we can do:

```c
DenyType t = PTR_TO_INT(hashmap_get(denylist, st));
```

and if the inode is not in the list the hashmap_get() would return NULL, and thus it would get mapped to DENY_DONT, and could be nicely handled.